src/test/java/seedu/address/storage/JsonTaskStorageTest.java

```diff
@@ -0,0 +1,98 @@
+package seedu.address.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalTasks.TASK_FUEL;
+import static seedu.address.testutil.TypicalTasks.TASK_GARLIC;
+import static seedu.address.testutil.TypicalTasks.getTypicalTaskList;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.model.ReadOnlyTaskList;
+import seedu.address.model.TaskList;
+
+public class JsonTaskStorageTest {
+    private static final Path TEST_DATA_FOLDER = Paths.get("src", "test", "data", "JsonTaskStorageTest");
+
+    @TempDir
+    public Path testFolder;
+
+    @Test
+    public void readTaskList_nullFilePath_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> readTaskList(null));
+    }
+
+    private java.util.Optional<ReadOnlyTaskList> readTaskList(String filePath) throws Exception {
+        return new JsonTaskStorage(Paths.get(filePath)).readTaskList(addToTestDataPathIfNotNull(filePath));
```

Comment:
"Paths.get(filePath)" seems to be incorrect, since it is missing the folder path

Reply:
It is fine here. `addToTestDataPathIfNotNull(filePath)` will already have resolved the folder path given the `filePath`. Also notice that `JsonTaskStorage` constructor takes in a `filePath` as we do not want to have a situation where we hardcode the `filePath` in `JsonTaskStorage`. This applies to this test function as well. It is identical to `JsonAddressBookTest`.

![image](https://user-images.githubusercontent.com/56034480/195286653-15a2e686-7128-4637-a125-906f3ae00f56.png)